Userland/Applications/HexEditor/HexEditorWidget.cpp

```diff
@@ -356,46 +358,55 @@ void HexEditorWidget::update_inspector_values(size_t position)
     size_t valid_bytes;
     utf8_view.validate(valid_bytes);
     if (valid_bytes == 0)
         value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF8, "");
-    else
-        value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF8, utf8_view.unicode_substring_view(0, 1).as_string());
+    else {
+        auto utf8_char = utf8_view.unicode_substring_view(0, 1).as_string();
+        auto utf8_display_char = make_display_string(utf8_char).release_value_but_fixme_should_propagate_errors().to_deprecated_string();
+        value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF8, utf8_display_char);
+    }
 
+    // FIXME: The byte read count doesn't need to be even for UTF-16 to be valid
     if (byte_read_count % 2 == 0) {
         Utf16View utf16_view { ReadonlySpan<u16> { reinterpret_cast<u16 const*>(&unsigned_64_bit_int), 4 } };
         size_t valid_code_units;
         utf16_view.validate(valid_code_units);
         if (valid_code_units == 0)
             value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF16, "");
-        else
-            value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF16, utf16_view.unicode_substring_view(0, 1).to_deprecated_string().release_value_but_fixme_should_propagate_errors());
+        else {
+            auto utf16_char = utf16_view.unicode_substring_view(0, 1).to_utf8().release_value_but_fixme_should_propagate_errors();
+            auto utf16_display_char = make_display_string(StringView { utf16_char }).release_value_but_fixme_should_propagate_errors().to_deprecated_string();
+            value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF16, utf16_display_char);
+        }
     } else {
         value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF16, "");
     }
 
     auto selected_bytes = m_editor->get_selected_bytes();
 
-    auto ascii_string = DeprecatedString { ReadonlyBytes { selected_bytes } };
-    value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::ASCIIString, ascii_string);
+    auto ascii_string = StringView { selected_bytes };
+    auto ascii_display_string = make_display_string(ascii_string).release_value_but_fixme_should_propagate_errors().to_deprecated_string();
+    value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::ASCIIString, ascii_display_string);
 
     Utf8View utf8_string_view { ReadonlyBytes { selected_bytes } };
     utf8_string_view.validate(valid_bytes);
     if (valid_bytes == 0)
         value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF8String, "");
-    else
-        // FIXME: replace control chars with something else - we don't want line breaks here ;)
-        value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::UTF8String, utf8_string_view.as_string());
+    else {
```

Comment:
https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#omission-of-curly-braces-from-statement-blocks